onnxruntime/core/session/inference_session.cc

```diff
@@ -144,7 +144,7 @@ static bool AreAllComputeNodesAssignedToCudaEp(const Graph& graph) {
 
     // Empty node provider means CPU EP
     if (!node_provider.empty() &&
-        node_provider != kCudaExecutionProvider &&
+        !(node_provider == kCudaExecutionProvider || node_provider == kRocmExecutionProvider) &&
         node_provider != kCpuExecutionProvider) {
       nodes_on_cpu_and_cuda_eps_only = false;
```

Comment:
Nit: Var name to reflect rocm ?

Reply:
Sorry I don't understand your nit?  The new line has kRocmExecutionProvider to reflect rocm?

Reply:
Like `nodes_on_cpu_cuda_rocm_eps_only`